### PR TITLE
additional tables - fixes

### PR DESCRIPTION
**Pull request by lhendey**

### CODE REVIEW

**Comment by lhendey** [comment]
@rpitingolo - the % of AMI for the DC HPTF categories are not adding to 100%. can you double check?

**Comment by rpitingolo** [comment]
@lhendey OK, I see it. I'll check the code.

**Comment by rpitingolo** [comment]
@lhendey I think the problem is that the code tops out at 8 person HHs. So numprec=8 should really be numprec>=8.

Do you have a branch open already or do you want me to do it really quick?

**Comment by lhendey** [comment]
@rpitingolo i do have a branch open (on this pull request) analysis-#54 - can you adjust the code?

**Comment by rpitingolo** [comment]
@lhendey yeah I just pushed that commit.

**Comment by lhendey** [comment]
@rpitingolo did the vacancy rate not get output or did i miss it?

**Comment by rpitingolo** [comment]
@lhendey you're not missing it. Looks like we created it in the code but then for some reason it didn't make it to the output step. I'll check on this now to see what's involved in getting it outputted.

**Comment by rpitingolo** [comment]
@lhendey there's a problem in the proc summary that adds up the number of vacant and total units. I've been  troubleshooting for the past half hour but haven't gotten it yet. I don't think I'm going to have it figured out before I have to leave in 15 min. I can take a look again later tonight. Sorry.

**Comment by rpitingolo** [comment]
@lhendey this is the proc summary I'm stuck on:

proc summary data = pretables_withvac_wts;
	class ward2012 largehh;
	var &cvars. multigen grouphouse studenthouse righthoused overhoused underhoused 
		 vacant allhh_withvac;
	weight whhwt;
	output out = table2b_pre sum=;
run;

**Comment by rpitingolo** [comment]
@lhendey OK, I think I finally got it. THe new weights created some issues with how the vacancy stuff was coded. I saved the updated 2B sheet on Box.

**Comment by lhendey** [comment]
@rpitingolo - how come there is a vacancy rate on the person sheets? (for small households).

**Comment by rpitingolo** [comment]
@lhendey it's treating the missing bedrooms as less than 3  (I should have coded it 0 < bedrooms < 3 or just not calculated it for the person sheets). You can ignore or delete it since it's functionally a vacancy rate for all households.

File: Prog/LgUnit - Ipums Table 2B.sas [hunk 1]
```diff
@@ -147,6 +147,16 @@ data vacdata;
 			else if rent > 0 then vac = 1;
 		if vac = 1;
 	%end;
+
+	%else %if &ten. = ALL %then %do;
```
Comment (lhendey) [inline Prog/LgUnit - Ipums Table 2B.sas]: @rpitingolo - i'm not sure i'm following this code here. why if there is a value greater than 0 or rent greater than 0 would it also be vac=1? (also the rent line repeats rent? is that just a flag to include the observation in the dataset (vs assigning a value to a variable)?

**Comment by lhendey** [comment]
@rpitingolo - also we should have data for the HUD inc unit variables for vacant properties.  when i looked at the last pretables_collapse its missing (which i think should be renter units?)

File: Prog/LgUnit - Ipums Table 2B.sas (hunk 1 above)
Comment (rpitingolo) [reply Prog/LgUnit - Ipums Table 2B.sas]: @lhendey I'll take a look. I think I added this last Friday when trying to get the vacancy rate stuff working again.

**Comment by rpitingolo** [comment]
@lhendey looks like this line of code was screwing up the vacant unit calculations. `if gq in (1,2);` I'm fixing it now.

**Comment by rpitingolo** [comment]
@lhendey the person-level stuff looks OK. In the pretables datastep I added this code:

```
	%if &largedef. = PERSON %then %do;
	if vac ^= 1 then do;
		if numprec >= 4 then largehh = 1;
		else largehh = 0;
	end;
	%end;
```
So for the rows where vac = 1 it's leaving it missing in the largehh variable. Then in the proc summaries we're using largehh as the class, so anywhere it's missing it's not getting summarized. Let me know if you have any more questions about how it works.

**Comment by ptatian** [comment]
@lhendey @rpitingolo Since we've completed a draft, suggest merging this pull request and opening new issue(s) for any further revisions we need to make.

**Pull request merged**